components/lcd/esp_lcd_ra8875/esp_lcd_ra8875.c

```diff
@@ -0,0 +1,398 @@
+    ra8875->base.set_gap = panel_ra8875_set_gap;
+    ra8875->base.mirror = panel_ra8875_mirror;
+    ra8875->base.swap_xy = panel_ra8875_swap_xy;
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    ra8875->base.disp_off = panel_ra8875_disp_on_off;
+#else
+    ra8875->base.disp_on_off = panel_ra8875_disp_on_off;
+#endif
+    *ret_panel = &(ra8875->base);
+    ESP_LOGD(TAG, "new ra8875 panel @%p", ra8875);
+
+    return ESP_OK;
+
+err:
+    if (ra8875) {
+        if (panel_dev_config->reset_gpio_num >= 0) {
+            gpio_reset_pin(panel_dev_config->reset_gpio_num);
+        }
+        free(ra8875);
+    }
+    return ret;
+}
+
+static esp_err_t panel_ra8875_del(esp_lcd_panel_t *panel)
+{
+    ra8875_panel_t *ra8875 = __containerof(panel, ra8875_panel_t, base);
+
+    if (ra8875->reset_gpio_num >= 0) {
+        gpio_reset_pin(ra8875->reset_gpio_num);
+    }
+    ESP_LOGD(TAG, "del ra8875 panel @%p", ra8875);
+    free(ra8875);
+    return ESP_OK;
+}
+
+static esp_err_t panel_ra8875_reset(esp_lcd_panel_t *panel)
+{
+    ra8875_panel_t *ra8875 = __containerof(panel, ra8875_panel_t, base);
+    esp_lcd_panel_io_handle_t io = ra8875->io;
+
+    // perform hardware reset
+    if (ra8875->reset_gpio_num >= 0) {
+        gpio_set_level(ra8875->reset_gpio_num, ra8875->reset_level);
+        vTaskDelay(pdMS_TO_TICKS(10));
+        gpio_set_level(ra8875->reset_gpio_num, !ra8875->reset_level);
+        vTaskDelay(pdMS_TO_TICKS(10));
+    } else { // perform software reset
+        esp_lcd_panel_io_tx_param(io, LCD_CMD_SWRESET, NULL, 0);
+        vTaskDelay(pdMS_TO_TICKS(20)); // spec, wait at least 5ms before sending new command
+    }
+
+    return ESP_OK;
+}
+
+typedef struct {
+    uint16_t cmd;
+    uint8_t data[16];
+    uint8_t data_bytes; // Length of data in above data array; 0xFF = end of cmds.
+} lcd_init_cmd_t;
+
+static const lcd_init_cmd_t vendor_specific_init[] = {
+    /* PLL init */
+    {0x88, {0x0b}, 1},
+    {0x89, {0x01}, 1},
+
+    /* Pixel Clock */
+    {0x04, {0x81}, 1},
+
+    {0x15, {0x03}, 1},
+    {0x16, {0x03}, 1},
+    {0x17, {0x02}, 1},
+    {0x18, {0x00}, 1},
+
+    {0x1b, {0x14}, 1},
+    {0x1c, {0x00}, 1},
+    {0x1d, {0x06}, 1},
+    {0x1e, {0x00}, 1},
+    {0x1f, {0x01}, 1},
+
+    {0x30, {0x00}, 1},
+    {0x31, {0x00}, 1},
+    {0x34, {0x1f}, 1},
+    {0x35, {0x03}, 1},
+
+    {0x32, {0x00}, 1},
+    {0x33, {0x00}, 1},
+    {0x36, {0xdf}, 1},
+    {0x37, {0x01}, 1},
+
+    /* Backlight PWM1 - settings */
+    {0x8a, {0x95}, 1},
+    /* Backlight PWM1 - duty */
+    {0x8b, {0x05}, 1},
+
+    /* End */
+    {0, {0}, 0xff},
+};
+
+static void panel_ra8875_wait(esp_lcd_panel_t *panel)
+{
+    ra8875_panel_t *ra8875 = __containerof(panel, ra8875_panel_t, base);
+    if (ra8875->wait_gpio_num >= 0) {
+        uint64_t start = esp_timer_get_time();
+        while (gpio_get_level(ra8875->wait_gpio_num) == 0 && (esp_timer_get_time() - start) < ESP_RA8875_TIMEOUT_MS);
```

Comment:
Although I like this easy solution with esp_timer, I have two points;
* esp_timer works with micro seconds resolution (your macro assumes miliseconds)
* there should be error returned in case of timeout!